zeppelin-web-angular/e2e/tests/notebook/keyboard/notebook-keyboard-shortcuts.spec.ts

```diff
@@ -0,0 +1,867 @@
+      await keyboardPage.focusCodeEditor(0);
+      await keyboardPage.pressClearOutput();
+
+      // Then: Output should be cleared
+      const resultLocator = keyboardPage.getParagraphByIndex(0).locator('[data-testid="paragraph-result"]');
+      await expect(resultLocator).not.toBeVisible();
+    });
+  });
+
+  test.describe('ParagraphActions.Link: Control+Alt+W', () => {
+    test('should trigger link paragraph with Control+Alt+W', async () => {
+      // Given: A paragraph with content
+      await keyboardPage.focusCodeEditor();
+      await keyboardPage.setCodeEditorContent('%md\n# Link Test\nTesting link paragraph functionality');
+
+      // Verify content was set correctly
+      const initialContent = await keyboardPage.getCodeEditorContent();
+      expect(initialContent.replace(/\s+/g, ' ')).toContain('link');
+
+      // When: User presses Control+Alt+W
+      await keyboardPage.pressLinkParagraph();
+
+      // Then: Link action should be triggered (verify basic functionality)
+      await keyboardPage.page.waitForTimeout(1000);
+      const content = await keyboardPage.getCodeEditorContent();
+      expect(content.length).toBeGreaterThan(0);
+      expect(content).toMatch(/link|test/i);
+
+      // Verify system remains functional
+      const paragraphCount = await keyboardPage.getParagraphCount();
+      expect(paragraphCount).toBeGreaterThanOrEqual(1);
+    });
+  });
+
+  // ===== PARAGRAPH WIDTH SHORTCUTS =====
+
+  test.describe('ParagraphActions.ReduceWidth: Control+Shift+-', () => {
+    test('should reduce paragraph width with Control+Shift+-', async () => {
+      // Given: A paragraph
+      await keyboardPage.focusCodeEditor();
+      await keyboardPage.setCodeEditorContent('%python\nprint("Test width reduction")');
+
+      const initialWidth = await keyboardPage.getParagraphWidth(0);
+
+      // When: User presses Control+Shift+-
+      await keyboardPage.pressReduceWidth();
+
+      // Then: Paragraph width should change
+      const finalWidth = await keyboardPage.getParagraphWidth(0);
+      expect(finalWidth).not.toBe(initialWidth);
+    });
+  });
+
+  test.describe('ParagraphActions.IncreaseWidth: Control+Shift+=', () => {
+    test('should increase paragraph width with Control+Shift+=', async () => {
+      // Given: A paragraph
+      await keyboardPage.focusCodeEditor();
+      await keyboardPage.setCodeEditorContent('%python\nprint("Test width increase")');
+
+      // First, reduce width to ensure there's room to increase
+      await keyboardPage.pressReduceWidth();
+      await keyboardPage.page.waitForTimeout(500); // Give UI a moment to update after reduction
+
+      const initialWidth = await keyboardPage.getParagraphWidth(0);
+
+      // When: User presses Control+Shift+=
+      await keyboardPage.pressIncreaseWidth();
+
+      // Then: Paragraph width should change
+      const finalWidth = await keyboardPage.getParagraphWidth(0);
+      expect(finalWidth).not.toBe(initialWidth);
```

Comment:
The same as the above.

```suggestion
      expect(finalWidth).toBeGreaterThan(initialWidth);
```